src/session/user.rs

```diff
@@ -294,7 +294,12 @@ impl User {
 
         gtk::ClosureExpression::new(
             move |args| -> String {
-                let first_name = args[1].get::<&str>().unwrap();
+                let mut first_name = args[1].get::<&str>().unwrap();
+
+                // Name shouldn't ever be "" unless it's deleted account
```

Comment:
Actually we don't know that. To know if the user is "deleted" we have to check if its type is `Deleted`, so let's do that instead. In this case, since we're using expressions, you have to create a new expression for the user type and then pass it to the `ClosureExpression`. You can look at the `User::first_name_expression()` to better understand how to do that.